Source/WebCore/dom/CustomElementDefaultARIA.cpp

```diff
@@ -91,8 +91,10 @@ RefPtr<Element> CustomElementDefaultARIA::elementForAttribute(const Element& thi
 
     RefPtr<Element> result;
     std::visit(WTF::makeVisitor([&](const AtomString& stringValue) {
-        if (thisElement.isInTreeScope())
-            result = thisElement.treeScope().getElementById(stringValue);
+        if (thisElement.isInTreeScope()) {
+            RefPtr elementForId = thisElement.treeScope().getElementById(stringValue);
+            result = elementForId ? elementForId->deepShadowRootReferenceTargetOrSelf() : nullptr;
```

Comment:
This code is duplicated in getElementByIdWithReferenceTarget in AXObjectCache.cpp.
Maybe we should add a helper function to TreeScope like `TreeScope::elementByIdRespectingReferenceTarget` or something?